### PR TITLE
feat: localize strings

### PR DESCRIPTION
**Pull request by audrey-kho**
## Description

String | L10n Status
-- | --
You are about to update {}, last modified on {} by {}. Any changes made will overwrite the current content.** | ✅
Select a source | ✅
Select from Adobe Target | ✅
Select from CSV file | ✅
Today | ✅
Yesterday | ✅
(other date labels in `FormatHelper.js`) | ✅
The characters {, }, and " are reserved and can't be used within quoted text values. Please remove or replace these characters and try again. | ✅
Variation created | ✅
An error occurred while sending feedback | ✅
An error occurred while generating results | ✅
The response was filtered due to the prompt triggering Generative AI's content management policy. Please modify your prompt and retry. | ✅
Generative AI's request timed out. Please try again by reducing the number of variations. | ✅
Generative AI's rate limit exceeded. Please wait one minute and try again. | ✅
The access token is not valid | ✅
The access token is not provided | ✅
Failed to fetch profile | ✅
Profile does not have access to the product | ✅
Apologies, it appears that you lack permission to use this feature.Please try selecting a different organization or contact your Administrator to request access. | ✅
Oops! We've encountered an unexpected error. Please try again later. | ✅
Information | n/a (Spectrum component `aria-label`)
Informations | n/a (Spectrum component `aria-label`)

**Note: Snippets of this complex string are not being translated properly for now (will be resolved when human translations are done)

## Related Issue

[https://jira.corp.adobe.com/browse/SITES-21934](https://jira.corp.adobe.com/browse/SITES-21934)
## How Has This Been Tested?
**English**: https://experience-qa.adobe.com/?devMode=true&shell_source=workspace&workspace=bulkl10n&shell_ims=prod&shell_locale=en-us#/aem/generate-variations/

**Chinese**: https://experience-qa.adobe.com/?devMode=true&shell_source=workspace&workspace=bulkl10n&shell_ims=prod&shell_locale=zh-cn#/aem/generate-variations/

**Japanese**: https://experience-qa.adobe.com/?devMode=true&shell_source=workspace&workspace=bulkl10n&shell_ims=prod&shell_locale=ja-jp#/aem/generate-variations/

**Spanish**: https://experience-qa.adobe.com/?devMode=true&shell_source=workspace&workspace=bulkl10n&shell_ims=prod&shell_locale=es-es#/aem/generate-variations/

**French**: https://experience-qa.adobe.com/?devMode=true&shell_source=workspace&workspace=bulkl10n&shell_ims=prod&shell_locale=fr-fr#/aem/generate-variations/

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by irenelagno** [comment]
Missed date localization in the side panel:

File: web-src/src/components/ContentFragmentExportButton.js [hunk 1]
```diff
@@ -64,10 +66,20 @@ export function ContentFragmentExportButton({ variant }) {
           const url = `https://experience.adobe.com/?repo=${new URL(aemService.getHost()).host}#/aem/cf/editor/editor${contentFragment.fragment.path}`;
           window.open(url, '_blank');
         }
-        ToastQueue.positive('Variation created', { timeout: 1000 });
```
Comment (irenelagno) [inline web-src/src/components/ContentFragmentExportButton.js:67]: should we wrap it to formatMessage method?

File: web-src/src/helpers/FormatHelper.js [hunk 2]
```diff
@@ -68,3 +68,16 @@ export function newGroupingLabelGenerator() {
     return null;
   };
 }
+
+export function getErrorCodeSubstring(str) {
+  const result = str?.match(/{{(.*?)}}/);
+  return result ? result[1] : undefined;
+}
+
+export function createToastErrorMessage(message, localizedError) {
```
Comment (irenelagno) [inline web-src/src/helpers/FormatHelper.js:77]: Should we name this method more abstract, like getMessageWithLocalization or handleLocalizedResponse?
Please add a description to this method to explain its purpose.

File: web-src/src/helpers/FormatHelper.js [hunk 3]
```diff
@@ -68,3 +68,16 @@ export function newGroupingLabelGenerator() {
     return null;
   };
 }
+
+export function getErrorCodeSubstring(str) {
```
Comment (irenelagno) [inline web-src/src/helpers/FormatHelper.js:72]: I'd recommend to change this method return format to be boolean and rename it to smth isLocalizationAvailable.

**Review by irenelagno — CHANGES_REQUESTED** [review]

**Comment by irenelagno** [comment]
@audrey-kho here is an example on hoe to localize data https://github.com/adobe/aem-genai-assistant/blob/main/web-src/src/components/PromptResultCard.js#L215

You can look into the PromptResult card as a reference.

File: web-src/src/components/PromptSessionSideView.js [hunk 4]
```diff
@@ -88,7 +90,14 @@ export function PromptSessionSideView({
         ? <Flex UNSAFE_className={styles.promptFlexItems} UNSAFE_style={{ borderBottom: '1px solid rgb(224, 224, 224)' }} direction={'column'} justifyContent={'stretch'} alignItems={'stretch'} gridArea={'info'}>
           <Flex UNSAFE_style={{ borderRadius: '8px', background: '#E0F2FF', padding: '10px' }} gap={'size-100'} alignItems={'center'}>
             <GenAIIcon />
-            <Text UNSAFE_className={styles.promptName}>{currentSession.name ?? 'Empty'}</Text>
+            <Text UNSAFE_className={styles.promptName}>{`${currentSession.name.split(' ')[0]} ${new Date(currentSession.name).toLocaleString(user.locale, {
```
Comment (lenaorobei) [inline web-src/src/components/PromptSessionSideView.js:93]: I recommend moving this to a constant.
Why is this needed `${currentSession.name.split(' ')[0]`?

File: actions/AuthNAction.js [hunk 5]
```diff
@@ -64,12 +64,15 @@ async function getAccessToken(params) {
   ) {
     const accessToken = params.__ow_headers.authorization.substring('Bearer '.length);
     // Validate the access token
+
+    // The error identifiers (enclosed in double curly braces) will be replaced with the actual
+    // error messages after being processed for localization on the frontend
     if (!await isValidToken(imsEndpoint, clientId, accessToken)) {
-      throw new Error('The access token is not valid');
+      throw new Error('{{invalidAccessToken}}');
```
Comment (slitviachenko) [inline actions/AuthNAction.js:71]: In general, API responses should be self-explanatory, at least for the developers who use them, and optionally provide some additional data to more closely identify error responses. This could include a link to documentation or an internal error code that might be mapped to the error. Clients (our web app in this case) might use this information to properly localize and format a message shown to end users.

I believe we agreed to use this [approach](https://github.com/adobe/aem-genai-assistant/pull/282#discussion_r1621457969). In this PR, as I understand, the returned error message will be:
```
{
  "error": "{{invalidAccessToken}}"
}
```
?

Comment (askayastha22) [reply actions/AuthNAction.js:71]: Yes, I suggested this method as the original error message needs to live in the frontend code for the localization to work correctly.

**Review by askayastha22 — APPROVED** [review]
LGTM

**Review by lenaorobei — APPROVED** [review]
